qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -812,6 +856,17 @@ def _evaluate_double_pulse_function(self, fit_function: IFunction, output_worksp
         alg.setProperty("Output", output_workspace)
         alg.execute()
 
+    def _create_guess_range_workspace(self, x_range: np.array, output_workspace: str, store_in_ads: bool = True) -> Workspace:
+        """Create a workspace with the requested fit range."""
+        alg = AlgorithmManager.create("CreateWorkspace")
+        alg.initialize()
+        alg.setAlwaysStoreInADS(store_in_ads)
+        alg.setProperty("DataX", x_range)
+        alg.setProperty("DataY", x_range)
+        alg.setProperty("OutputWorkspace", output_workspace)
+        alg.execute()
+        return alg.getProperty('OutputWorkspace').value
```

Comment:
Can we use this utility function which is in the algorithm utils file?
https://github.com/mantidproject/mantid/blob/cffb8050f43a522a2ff059a7834b0e4f59aa1075/qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/utilities/algorithm_utils.py#L377

Reply:
We don;t want te workspace stored in the ADS so I need to set that in the algorithm

Reply:
@Harrietbrown does it still work with workspace history? 

We could modify the  function to be
```
def run_create_workspace(x_data, y_data, name, store_in_ADS=True)
```

Reply:
I've tried it, but the function returns a workspace name not a workspace itself. so it can;t be used in the Evaluate function since it is not in the ADS.

Reply:
Would something like this work?
```
def run_create_workspace(x_data, y_data, name, store_in_ADS=True, get_object: bool = False)
```
and then
```
if get_object:
    return alg.getProperty('OutputWorkspace').value 
else:
    return alg.getProperty('OutputWorkspace').valueAsStr
```

Reply:
Or maybe we don't need `get_object` and we can just do:
```
if not store_in_ADS:
    return alg.getProperty('OutputWorkspace').value 
else:
    return alg.getProperty('OutputWorkspace').valueAsStr
```

Reply:
As typehints are becoming more and more used in our codebase I am hesitent to implement a change that would make a python function return different types.